### DISCUSSION TITLE
Override Entry's default Augmentation Method

### DISCUSSION THREAD

**Discussion by barnabas-szekeres** [statamic/cms#5076] [Q&A] [answered]
Hi All,

Is there any good solution to extend / override the Entry repositories default augmentation logic? We have some custom controller which run queries on Entires. For example:

```
$shows = Entry::query()
    ->where('collection', 'shows')
    ->where('published', true)
    ->get();

$shows->toAugmentedArray(['title', 'slug', 'thumbnail']);
```
            
However the process will takes a very long time (2-3sec!) because of the the augmentation of the `thumbnail` property (which is an S3 asset).

I read several times the related documentation, but I didn't find answer. :( 

https://statamic.dev/extending/augmentation#augmentable

**Comment by jasonvarga** [comment, marked as answer]
I'm not sure why it would take 2-3 seconds for that. Perhaps your stache watcher is enabled (in `config/statamic/stache.php`), which would improve developer experience at the expense of performance. It's fine to have it enabled in development, but turn it off in production.

Anyway, it's not the Entry class that would control it, it's each fieldtype.

You could just _not_ augment that field, then put the raw value in:

```php
$arr = $show->toAugmentedArray(['title', 'slug']);
$arr['thumbnail'] = $show->value('thumbnail');
```

or since you're only using those few things, just build your own array:

```php
$arr = [
  'title' => $show->value('title'),
  'slug' => $show->slug(),
  'thumbnail' => $show->value('thumbnail'),
];
```
